### ISSUE TITLE
Errors from `pip install -e .`

### ISSUE THREAD

**Issue by sajith** [atlanticwave-sdx/datamodel#29]
README suggests using `pip install -e .`, but that did not work for me:

```
$ pip install -e .
Obtaining file:///home/sajith/projects/renci/aw-sdx/datamodel
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [14 lines of output]
      error: Multiple top-level packages discovered in a flat-layout: ['schemas', 'sdxdatamodel'].
      
      To avoid accidental inclusion of unwanted files or directories,
      setuptools will not proceed with this build.
      
      If you are trying to create a single distribution with multiple packages
      on purpose, you should not rely on automatic discovery.
      Instead, consider the following options:
      
      1. set up custom discovery (`find` directive with `include` or `exclude`)
      2. use a `src-layout`
      3. explicitly set `py_modules` or `packages` with a list of names
      
      To find more information, look for "package discovery" on setuptools docs.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

**Comment by sajith** [comment]
Closed in PR #35.

**Comment by sajith** [comment]
Somebody seems to have forgotten to add some details that could have been useful to reproduce the issue that they reported. :^)

(Details such as: what version of Python and pip, what commit, what OS...)

I am coming back to this today, and I am having trouble reproducing this.  I tried reproducing this on Fedora 36 and a Debian 11.3 container.  I will probably add a step in CI to test that `pip install -e .` works as claimed, and declare victory if it does.

**Comment by sajith** [comment]
I am working on setting up CI for sdx-controller.  Turns out hat sdx-controller's tests need datamodel, but we can't install datamodel on CI. The current work in progress is [here](https://github.com/sajith/sdx-controller/runs/8122483459.

And here's the error message:


```
================================== log start ===================================
Collecting git+https://github.com/atlanticwave-sdx/datamodel (from -r /home/runner/work/sdx-controller/sdx-controller/test-requirements.txt (line 9))
  Cloning https://github.com/atlanticwave-sdx/datamodel to /tmp/pip-req-build-oasjabk7
  Running command git clone --filter=blob:none --quiet https://github.com/atlanticwave-sdx/datamodel /tmp/pip-req-build-oasjabk7
  Resolved https://github.com/atlanticwave-sdx/datamodel to commit 656e572308edcfb93fada6cc58e3cb87e9407895
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'error'
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [[14](https://github.com/sajith/sdx-controller/runs/8122483459?check_suite_focus=true#step:8:14) lines of output]
      error: Multiple top-level packages discovered in a flat-layout: ['schemas', 'sdxdatamodel'].
      
      To avoid accidental inclusion of unwanted files or directories,
      setuptools will not proceed with this build.
      
      If you are trying to create a single distribution with multiple packages
      on purpose, you should not rely on automatic discovery.
      Instead, consider the following options:
      
      1. set up custom discovery (`find` directive with `include` or `exclude`)
      2. use a `src-layout`
      3. explicitly set `py_modules` or `packages` with a list of names
      
      To find more information, look for "package discovery" on setuptools docs.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed
× Encountered error while generating package metadata.
╰─> See above for output.
note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
=================================== log end ====================================
```

**Comment by sajith** [comment]
Should be fixed with PR #40.

**Comment by sajith** [comment]
Should be fixed with PR #40.
